File: ruby/day01/part-2/answer.rb
file = File.open("../part-1/input.txt")

def calculate_fuel(mass, running_count = 0)
  fuel = (mass / 3) - 2
  return running_count if fuel <= 0
  calculate_fuel(fuel, running_count + fuel)
end

fuel_req = 0
file.each do |mass|
  fuel_req += calculate_fuel(mass.to_i)
end

puts fuel_req
# 5097039

File: elixir/day01/part-1/answer.exs
# file = File.read("./input.txt")

sample = File.read!("./input.txt") |> String.split |> Enum.map(&String.to_integer/1)
IO.inspect sample

# IO.puts Enum.reduce(contents, fn x, acc -> acc + Integer.floor(String.to_integer(x) / 3) - 2 end) 

# total_fuel = 0 
# file.each do |mass|
#   fuel_req = (mass.to_i / 3) - 2
#   total_fuel += fuel_req
# end

# puts total_fuel
# 3399947